osu.Server.Spectator/Entities/UserHubEntities.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Server.Spectator.Hubs;
+
+namespace osu.Server.Spectator.Entities
+{
+    public abstract class UserHubEntities<TUserState>
+        where TUserState : ClientState
+    {
+        public readonly EntityStore<TUserState> ActiveUsers = new EntityStore<TUserState>();
+    }
```

Comment:
Do these need to exist at all? The `EntityStores` can be registered directly, by specifying the constructed generic types (`EntityStore<SpectatorClientState>` and `EntityStore<MultiplayerClientState>`), no?

Reply:
Definitely, I just thought it would be a better idea to encapsulate things rather than caching each `EntityStore` per hub, though it may be an overcomplicated thing so not sure anymore.

Reply:
The first example you give here is wrong. The multiplayer server shouldn't even have spectator states. I agree that these classes should not exist, though. Too many layers of abstraction.

Reply:
I wrote how it would look for both the `MultiplayerHub` constructor and the `SpectatorHub` constructor. I have just made these classes to shorten the dependencies through abstraction, but sure, can break that to direct `EntityStore`s if that's a better choice.